packages/grid/_modules_/grid/hooks/features/virtualization/useVirtualRows.ts

```diff
@@ -51,6 +50,7 @@ export const useVirtualRows = (
         if (!isEqual(oldState.rendering, currentRenderingState)) {
           oldState.rendering = currentRenderingState;
           stateChanged = true;
+          return { ...oldState };
```

Comment:
If we want to treat the state as immutable as possible, `.rendering` shouldn't mutate the previous instance.

Reply:
We just changed the state on line 51 so we create a new instance, otherwise it won't be taken into account

Reply:
why not do?
```suggestion
          stateChanged = true;
          return { ...oldState, rendering currentRenderingState };
```